### ISSUE TITLE
iOS Designer Requires an Active Connection to a Mac Server - Already Connected

### ISSUE THREAD

**Issue by decidedlygray** [xamarin/xamarin-macios#4112]
### Steps to Reproduce

1. Open a project.
2.  Use VS 2017 on a Windows PC to connect to a macOS build machine
3. Connection created successfully
4. Clean, build, deploy application. Works as expected - project runs on remote macOS
5.  Try to open a .storyboard file
6.  Opening the .storyboard files

### Expected Behavior

Expectation would be that IOS Designed on VS2017 Windows PC allows editing of the .storyboard file

### Actual Behavior

Even though connection to Mac is active, and builds/deployments to Mac work, the following error message shows up:
```
Something went wrong

The iOS Designer requires an active connection to a Mac server. Please connect to a Mac to enable the designer.
```

Additional Logs, with stacktraces, are here. 192.168.1.2 is the Windows PC, 192.168.1.3 is the Mac:

https://gist.github.com/decidedlygray/7b0bfe2ef35fac5d55671defc128c8dc

### Environment

I am using a Windows 10 machine, running the Visual Studio with info below, building against macOS.

Both machines were updated and rebooted last night. After updating is when this all started happening.
```
Microsoft Visual Studio Community 2017 
Version 15.7.2
VisualStudio.15.Release/15.7.2+27703.2018
Microsoft .NET Framework
Version 4.7.03056

Installed Version: Community

Application Insights Tools for Visual Studio Package   8.12.10405.1
Application Insights Tools for Visual Studio

ASP.NET and Web Tools 2017   15.0.40511.0
ASP.NET and Web Tools 2017

ASP.NET Core Razor Language Services   15.7.31476
Provides languages services for ASP.NET Core Razor.

Azure App Service Tools v3.0.0   15.0.40424.0
Azure App Service Tools v3.0.0

C# Tools   2.8.2-beta6-62916-08. Commit Hash: 2ad4aabc7a9dada097e54e544ebba48ab1c05074
C# components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Common Azure Tools   1.10
Provides common services for use by Azure Mobile Services and Microsoft Azure Tools.

Cookiecutter   15.7.18131.1
Provides tools for finding, instantiating and customizing templates in cookiecutter format.

JavaScript Language Service   2.0
JavaScript Language Service

JavaScript UWP Project System   2.0
JavaScript UWP Project System

Merq   1.1.19-rc (a4ffc1b)
Command Bus, Event Stream and Async Manager for Visual Studio extensions.

Microsoft Azure Tools   2.9
Microsoft Azure Tools for Microsoft Visual Studio 2017 - v2.9.10420.2

Microsoft JVM Debugger   1.0
Provides support for connecting the Visual Studio debugger to JDWP compatible Java Virtual Machines

Microsoft MI-Based Debugger   1.0
Provides support for connecting Visual Studio to MI compatible debuggers

Microsoft Visual Studio VC Package   1.0
Microsoft Visual Studio VC Package

Mono Debugging for Visual Studio   4.10.5-pre (ab58725)
Support for debugging Mono processes with Visual Studio.

NuGet Package Manager   4.6.0
NuGet Package Manager in Visual Studio. For more information about NuGet, visit http://docs.nuget.org/.

ProjectServicesPackage Extension   1.0
ProjectServicesPackage Visual Studio Extension Detailed Info

Python   15.7.18131.1
Provides IntelliSense, projects, templates, debugging, interactive windows, and other support for Python developers.

Python - Django support   15.7.18131.1
Provides templates and integration for the Django web framework.

Python - IronPython support   15.7.18131.1
Provides templates and integration for IronPython-based projects.

Python - Profiling support   15.7.18131.1
Profiling support for Python projects.

ResourcePackage Extension   1.0
ResourcePackage Visual Studio Extension Detailed Info

TypeScript Tools   15.7.20419.2003
TypeScript Tools for Microsoft Visual Studio

Visual Basic Tools   2.8.2-beta6-62916-08. Commit Hash: 2ad4aabc7a9dada097e54e544ebba48ab1c05074
Visual Basic components used in the IDE. Depending on your project type and settings, a different version of the compiler may be used.

Visual F# Tools 10.1 for F# 4.1   15.7.0.0.  Commit Hash: 56f97a8dd4353d897336941f3e644423b87c794f.
Microsoft Visual F# Tools 10.1 for F# 4.1

Visual Studio Code Debug Adapter Host Package   1.0
Interop layer for hosting Visual Studio Code debug adapters in Visual Studio

Visual Studio Tools for Universal Windows Apps   15.0.27703.2018
The Visual Studio Tools for Universal Windows apps allow you to build a single universal app experience that can reach every device running Windows 10: phone, tablet, PC, and more. It includes the Microsoft Windows 10 Software Development Kit.

VisualStudio.Mac   1.0
Mac Extension for Visual Studio

Windows Machine Learning Generator Extension   1.0
Windows Machine Learning Visual Studio Extension Detailed Info

Xamarin   4.10.0.448 (4373404db)
Visual Studio extension to enable development for Xamarin.iOS and Xamarin.Android.

Xamarin Designer   4.12.270 (82d750d12)
Visual Studio extension to enable Xamarin Designer tools in Visual Studio.

Xamarin.Android SDK   8.3.0.19 (HEAD/342b2ce96)
Xamarin.Android Reference Assemblies and MSBuild support.

Xamarin.iOS and Xamarin.Mac SDK   11.10.1.178 (408d357)
Xamarin.iOS and Xamarin.Mac Reference Assemblies and MSBuild support.
```

### Example Project (If Possible)

N/A

### Things I've Tried

- Restarting both machines
- Cleaning and rebuilding the solution on Windows VS2017 - works fine
- Cleaning temporary files on both machines, then rebooting, reconnecting, then trying to open .storyboard
- Opening XCode on Mac to make sure it works and can build a "hello world" app. Works fine
- Deploying project from Windows to Mac. Works as expected. App launches in emulator on Mac.
- Adding and removing the Mac from VS2017 connection settings. Did this multiple times. I can reconnect to the Mac server from VS2017 without issue
- Pulling hair and cursing to myself - doesn't seem to solve anything

**Any additional help or debugging steps would be greatly appreciated!!!**

> VS bug [#620469](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/620469), VS bug [#620470](https://devdiv.visualstudio.com/DevDiv/_workitems/edit/620470)

**Comment by decidedlygray** [comment]
Closing the loop on this:

In some of the stack traces I noticed it was referencing the Mono.framework on the Mac. I reinstalled VS on the Mac, updated everything to the latest stable.

That still didn't work (I was getting an error relating to updating storyboards). So I removed the connections from my Windows machine. Then wiped `/tmp` on the Mac. Then rebooted the Mac.

I am not able to edit storyboards.

Cheers!
